include/swift/Driver/Compilation.h

```diff
@@ -160,6 +160,15 @@ class Compilation {
   /// of date.
   bool EnableIncrementalBuild;
 
+  /// When true, emit duplicated compilation record file whose filename is
+  /// suffixed with '~moduleonly'.
+  ///
+  /// This compilation record is used by '-emit-module'-only incremental builds
+  /// so that module-only builds do not affect compilation record file for
+  /// normal builds, while module-only incremental builds are able to use
+  /// artifacts of normal builds if they are already up to date.
+  bool OutputCompilationRecordForModuleOnlyBuild = false;
+
```

Comment:
This is a great start! As someone new to the code, I am left with some questions, which may or may not be worth answering here: I *think* the idea is to allow a user to do a module-only build because it saves time over a build that emits both a module and an object file. But the compilation record file for the normal build isn't fine-grained enough to say that a .o file is out of date but a module file is current. Is this right? So the workaround is to create duplicate compilation records that only record currency info for the module files. Do I have this right? If so, where is the documented? Where should it be documented? Should this comment point to that place, if it does not include that info??